lib/file-view.coffee

```diff
@@ -19,7 +19,11 @@ class FileView extends HTMLElement
     @fileName.dataset.name = @file.name
     @fileName.dataset.path = @file.path
 
-    @fileName.classList.add(FileIcons.getService().iconClassForPath(@file.path))
+    iconClass = FileIcons.getService().iconClassForPath(@file.path, this)
```

Comment:
> Did you benchmark how file icons performs when issuing fs calls? How better it performs after putting a cache behind it?

It's not just that. Even if we cache the results for filesystem calls, we still have to monitor the paths to respond to filesystem changes. For instance, a path marked as a symlink might change, and be replaced with something that *isn't* symlinked. Caching is one thing, but having so many different listeners for separate files is another. And given the effort one would go to to reliably ascertain this sort of information, it feels kinda cruel to force package authors down that road.

Maybe just a flat object of bare minimum information, no references to package-defined classes, or anything?